src/components/BookConsultation/index.jsx

```diff
@@ -0,0 +1,245 @@
+              disabled={
+                formValues.ailment === "other"
+                  ? !formValues.otherAilment
+                  : !formValues.ailment
+              }
+            >
+              <MenuItem value="asap">ASAP</MenuItem>
+              <MenuItem value="later">Want to select date and time?</MenuItem>
+            </Select>
+          </FormControl>
+        </Grid>
+
+        {formValues.bookwhen === "later" && (
+          <Grid item>
+            <FormControl fullWidth>
+              <LocalizationProvider dateAdapter={AdapterDayjs}>
+                <DateTimePicker
+                  disablePast
+                  disabled={!formValues.bookwhen}
+                  label="Select a date and time of your choice"
+                  onChange={(value) => updateForm("datetime", value)}
+                />
+              </LocalizationProvider>
+            </FormControl>
+          </Grid>
+        )}
+
+        <Grid item>
+          <FormControl fullWidth>
+            <InputLabel htmlFor="hospital">
+              Select from the available hospitals
+            </InputLabel>
+            <Select
+              labelId="hospital"
+              onChange={(event) => updateForm("hospital", event.target.value)}
+              label="Select from the available hospitals"
+              disabled={
+                formValues.bookwhen === "later"
+                  ? !formValues.datetime
+                  : !formValues.bookwhen
+              }
+            >
+              <MenuItem value="hospital1">Hospital 1</MenuItem>
+              <MenuItem value="hospital2">Hospital 2</MenuItem>
+              <MenuItem value="hospital3">Hospital 3</MenuItem>
+              <MenuItem value="hospital4">Hospital 4</MenuItem>
+            </Select>
+          </FormControl>
+        </Grid>
+
+        <Grid item>
+          <FormControl fullWidth>
+            <InputLabel htmlFor="hospital">
+              Select from the available doctors
+            </InputLabel>
+            <Select
+              labelId="doctor"
+              disabled={!formValues.hospital}
+              onChange={(event) => updateForm("doctor", event.target.value)}
+              label="Select from the available doctors"
+            >
+              <MenuItem value="Doctor1">Doctor 1</MenuItem>
+              <MenuItem value="Doctor2">Doctor 2</MenuItem>
+              <MenuItem value="Doctor3">Doctor 3</MenuItem>
+              <MenuItem value="Doctor4">Doctor 4</MenuItem>
+            </Select>
+          </FormControl>
+        </Grid>
+
+        <Grid item>
+          <Grid container spacing={2}>
+            <Grid item xs={12} md={6}>
+              <Button
+                type="button"
+                color="success"
+                variant="contained"
+                disabled={!formValues.doctor}
+                fullWidth
+                onClick={() => {
+                  addToStorage(formValues);
+                  setFormAsCompleted(() => true);
```

Comment:
setFormAsCompleted(true)

Reply:
It is also another way of using state updation, this should be good.